### ISSUE TITLE
Alert doesn't reappear

### ISSUE THREAD

**Issue by venilnoronha** [vmware/clarity#2462]
**Select one ...** (check one with "x")

```
[x] bug
[ ] feature request
[ ] enhancement
```

### Expected behavior

The following snippet should show an alert whenever the selected element's `starships` count is 0.

```html
<clr-alert *ngIf="selected.starships.length === 0" [clrAlertSizeSmall]="true" [clrAlertType]="'danger'">
  <div class="alert-item">
    <span class="alert-text">
      {{selected.name}} has no starships.
    </span>
  </div>
</clr-alert>
```

### Actual behavior
The alert shows up initially; however, once the alert is closed via the X button, it doesn't reappear even if it's re-selected after selecting a bunch of other elements.

### Reproduction of behavior

https://stackblitz.com/edit/angular-clarity-july-18-weg5oc

### Environment details

* **Angular version:** 6.0.9

* **Clarity version:** 0.12.2

* **OS and version:** macOS 10.12.6

* **Browser:** Google Chrome Version 67.0.3396.99 (Official Build) (64-bit)

**Comment by gnomeontherun** [comment]
The alert doesn't open again because its not truly destroyed. The issue is around the *ngIf logic. Since it looks only at the `selected.starships.length` property, it doesn't really understand when the `selected`value changes. So if you close an alert, and select another option that doesn't have a starship, the alert remains closed because it hasn't been fully recycled (in other words, the content of the bindings has changed, but the *ngIf result has not so the alert is cached in its closed state). Instead, I added a second property to track the alertState, and actively reset it when you select a new option. 

Basically if you're going to reuse an alert like this, you need to reset the `clrAlertClosed` state to bring it back alive. You don't need to do that if the alert is _always_ destroyed, but in this case it remains in some cases.

I forked your stackblitz and fixed it up so it properly resets with each change.

https://stackblitz.com/edit/angular-clarity-july-18-6zprzb?file=app%2Fcomponents%2Fpeople%2Fpeople.component.html

Closing as this is an implementation/Angular issue, and feel free to post on StackOverflow if you have other implementation questions!

**Comment by venilnoronha** [comment]
@gnomeontherun thanks for the explanation!
